### PR TITLE
Assistance need preschool decisions - part 4

### PR DESCRIPTION
**Pull request by Joosakur**
#### Summary

First version of pdf generation, citizen views

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionCitizenController.kt [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                    if (
+                        decision.status !in
+                            listOf(
+                                AssistanceNeedDecisionStatus.ACCEPTED,
+                                AssistanceNeedDecisionStatus.REJECTED,
+                                AssistanceNeedDecisionStatus.ANNULLED
+                            )
+                    ) {
+                        throw NotFound(
+                            "Citizen can only view accepted and rejected assistance need decisions"
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionCitizenController.kt:78]: fix

File: service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt [hunk 2]
```diff
@@ -352,3 +357,111 @@ WHERE id = :id
...
+
+fun Database.Read.getAssistanceNeedPreschoolDecisionsForCitizen(
+    today: LocalDate,
+    userId: PersonId
+): List<AssistanceNeedPreschoolDecisionCitizenListItem> {
+    val childIds =
+        createQuery(
+                """
+        SELECT child_id FROM guardian WHERE guardian_id = :userId
+        UNION
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt:369]: ALL + set?

File: service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt [hunk 3]
```diff
@@ -352,3 +357,111 @@ WHERE id = :id
...
+
+    return childIds
+        .flatMap { childId -> getAssistanceNeedPreschoolDecisionsByChildId(childId) }
+        .filter {
+            it.status in
+                listOf(
+                    AssistanceNeedDecisionStatus.ACCEPTED,
+                    AssistanceNeedDecisionStatus.REJECTED,
+                    AssistanceNeedDecisionStatus.ANNULLED
+                )
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt:387]: `it.status.isDecided`

File: service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt [hunk 4]
```diff
@@ -352,3 +357,111 @@ WHERE id = :id
...
+): List<UnreadAssistanceNeedDecisionItem> {
+    // language=sql
+    val sql =
+        """
+        WITH children AS (
+            SELECT child_id FROM guardian WHERE guardian_id = :userId
+            UNION
+            SELECT child_id FROM foster_parent WHERE parent_id = :userId AND valid_during @> :today
+        )
+        SELECT ad.child_id, COUNT(ad.child_id) as count
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt:441]: improve

File: service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionController.kt [hunk 5]
```diff
@@ -333,6 +325,7 @@ class AssistanceNeedPreschoolDecisionController(
             .also { Audit.ChildAssistanceNeedPreschoolDecisionDelete.log(targetId = id) }
     }
 
+    // TODO: Unused endpoint?
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionController.kt:328]: copy paste from old implementation but didn't find any frontend using this, not sure if I missed something or can this be removed

**Review by Gekkio — APPROVED** [review]

**Pull request merged**